src/Sarif.UnitTests/SarifVersionOneToCurrentTests.cs

```diff
@@ -953,5 +951,168 @@ public void SarifTransformerTests_ToCurrent_NotificationExceptionWithStack()
+                            }
+                          ]
+		                }
+                      ],
+                      ""rules"": {
+                        ""C2001"": {
+                          ""id"": ""C2001"",
+                          ""shortDescription"": ""A variable was used without being initialized."",
+                          ""fullDescription"": ""A variable was used without being initialized. This can result in runtime errors such as null reference exceptions."",
+                          ""messageFormats"": {
+                            ""default"": ""Variable \""{0}\"" was used without being initialized.""
+                          }
+                        }
+                      }
+                    }
+                  ]
+                }";
+
+            SarifLog v2Log = TransformVersionOneToCurrent(v1LogText);
+
+            string v2LogText = JsonConvert.SerializeObject(v2Log, SarifTransformerUtilities.JsonSettingsV2);
+            string v2LogExpectedText =
+@"{
+  ""$schema"": ""http://json.schemastore.org/sarif-2.0.0"",
+  ""version"": ""2.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""logicalLocations"": {
+        ""collections::list::add"": {
+          ""name"": ""add"",
+          ""decoratedName"": ""?add@list@collections@@QAEXH@Z"",
+          ""parentKey"": ""collections::list"",
+          ""kind"": ""function""
+        },
+        ""collections::list"": {
+          ""name"": ""list"",
+          ""parentKey"": ""collections"",
+          ""kind"": ""type""
+        },
+        ""collections"": {
+          ""kind"": ""namespace""
+        }
+      },
+      ""results"": [
+        {
+          ""ruleId"": ""C2001"",
+          ""level"": ""error"",
+          ""ruleMessageId"": ""default"",
+          ""analysisTarget"": {
+            ""uri"": ""file:///home/buildAgent/src/collections/list.cpp""
+          },
+          ""locations"": [
+            {
+              ""physicalLocation"": {
+                ""fileLocation"": {
+                  ""uri"": ""file:///home/buildAgent/src/collections/list.h""
+                },
+                ""region"": {
+                  ""startLine"": 15,
+                  ""startColumn"": 9,
+                  ""endLine"": 15,
+                  ""endColumn"": 10,
+                  ""offset"": 254,
+                  ""length"": 1
+                }
+              },
+              ""fullyQualifiedLogicalName"": ""collections::list::add""
+            }
+          ],
+          ""suppressionStates"": [""suppressedExternally""],
+          ""baselineState"": ""existing"",
+          ""properties"": {
+            ""sarifv1/formattedRuleMessage"": {""formatId"":""default"",""arguments"":[""ptr""]},
+            ""sarifv1/snippet"": ""add_core(ptr, offset, val);\n    return;""
```

Comment:
You can put this in `region.snippet` of type `fileContents`.

Reply:
When you do that, let's make the properties of the region consistent with the length of the snippet.

Reply:
or locations[0] maybe

Reply:
Put the snippet in `locations[0].physicalLocation.region.snippet`, and make sure the properties of the `region` in your test log are consistent with the text of the `snippet`.